src/utils.php

```diff
@@ -138,11 +138,22 @@ function stripStringOverlap(string $a, string $b): string
  * in ascending order.
  *
  * @param array
- * @return array
+ * @return void
  */
 function sortUrisLevelOrder(&$uriList)
 {
-    usort($uriList, function ($a, $b) {
-        return substr_count(Uri\parse($a)['path'], '/') - substr_count(Uri\parse($b)['path'], '/');
+    // Parse URIs so we are not continually parsing them while sorting
+    $parsedUriList = array_map(function ($uri) {
+        return Uri\parse($uri);
+    }, $uriList);
```

Comment:
I'm sorry, I had a brain fart in my last review. I though `$a` and `$b` were the same, but they are not.
This is just micro optimisation \